src/core/utils/internal/isIterable.ts

```diff
@@ -1,11 +1,20 @@
+// Redefine these types; the three-parameter forms are only available in TypeScript 5.6 or later.
```

Comment:
Thank you for putting this in! Exactly what I was wondering about.

Reply:
I can also make these JSDoc comments so they're carried down; whatever you'd prefer.

Reply:
I'd mark them as `@todo` to remove these types once TS 5.6 is the minimal supported version. Not going to happen any time soon but still.

Reply:
Updated.